libgdf/src/types.cuh

```diff
@@ -0,0 +1,162 @@
+#ifndef GDF_CPPTYPES_H
+#define GDF_CPPTYPES_H
+
+#include <gdf/cffi/types.h>
+
+/* --------------------------------------------------------------------------*/
```

Comment:
Doxygen comments should _not_ have these separator lines before and after them. I don't know why people insist on adding them...

... if it's because of the documentation template: I specifically verified that these lines delimit template content from explanatory content, and are not part of the comments we need to put into the code.